Lib/test/test_zipfile/test_core.py

```diff
@@ -3348,6 +3348,12 @@ def test_data_offset_write_with_prefix(self):
             with zipfile.ZipFile(fp, "w") as zipfp:
                 self.assertEqual(zipfp.data_offset, 16)
 
+    def test_data_offset_append_with_bad_zip(self):
```

Comment:
Can we have a test where `data_offset` is None as well?

Reply:
I don't believe it ever could be `None` in append mode. Either it is set during reading the existing zip, or is set to the offset to the end of the existing data.

Reply:
I'm sorry, I meant a test where we could have the property to be None. Not necessarily in append mode.

Reply:
I believe that is covered by https://github.com/python/cpython/blob/main/Lib/test/test_zipfile/test_core.py#L3351-L3360, unless you had a different circumstance in mind?

Reply:
(In other words, the only path I can tell where it would be None is when `tell()` is not available in write mode)

Reply:
> unless you had a different circumstance in mind?

Ah I didn't see it because it used `assertIs(..., None)` instead of `assertIsNone`. Can you amend this check please? (other tests do use `assertIs(..., True)` or `assertIs(..., False)` because they likely want to have exact identity check and not just falsey or truthy values)

Reply:
Ah, I am not super familiar with unittest, so thank you for pointing that out! I've changed that test to use `self.assertIsNone()`.